scripts/build/mkversioninfo

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env sh
+set -eu
+
+: "${VERSION=}"
+: "${GITCOMMIT=}"
+: "${TARGET=}"
+
+# Create version quad for Windows of the form major.minor.patch.build
+VERSION_QUAD=$(printf "%s" "$VERSION" | sed -re 's/^([0-9.]*).*$/\1/' | sed -re 's/\.$//' | sed -re 's/^[0-9]+$/\0\.0/' | sed -re 's/^[0-9]+\.[0-9]+$/\0\.0/' | sed -re 's/^[0-9]+\.[0-9]+\.[0-9]+$/\0\.0/')
+
+# Generate versioninfo.json to be able to create a syso file which contains
+# Microsoft Windows Version Information and an icon using goversioninfo.
+# Example: https://github.com/josephspurrier/goversioninfo/blob/master/testdata/resource/versioninfo.json
+cat > ./cmd/docker/versioninfo.json <<EOL
+{
+  "FixedFileInfo":
+  {
+    "FileVersion": {
+      "Major": $(echo "$VERSION_QUAD" | cut -d. -f1),
+      "Minor": $(echo "$VERSION_QUAD" | cut -d. -f2),
+      "Patch": $(echo "$VERSION_QUAD" | cut -d. -f3),
+      "Build": $(echo "$VERSION_QUAD" | cut -d. -f4)
+    },
+    "FileFlagsMask": "3f",
+    "FileFlags ": "00",
+    "FileOS": "040004",
+    "FileType": "01",
+    "FileSubType": "00"
+  },
+  "StringFileInfo":
+  {
+    "Comments": "",
+    "CompanyName": "Docker, Inc.",
```

Comment:
Perhaps we should have a parameter to set this (in https://github.com/moby/moby/pull/42872, I used a `PACKAGER_NAME` arg for this). This way (e.g.) Microsoft / Azure packaging / building the binaries would be able to set their name.

Admitted; the purpose of this field is a bit unclear. Reading https://docs.microsoft.com/en-us/windows/win32/menurc/stringfileinfo-block, "company that produced the file" is unclear in what "produced" means (the binary? the source file?)

```
CompanyName
Company that produced the file—for example, "Microsoft Corporation" or "Standard
Microsystems Corporation, Inc." This string is required.
```

Reply:
/cc @cpuguy83 @tianon

Reply:
> "company that produced the file" is unclear in what "produced" means (the binary? the source file?)

I’m pretty sure it’s the binary, i.e. the file the user is looking at the properties in the Windows Explorer.

Reply:
Makes sense, yes; I think in that case, we should have a parameter, so that the party that does the packaging can set their name.

Reply:
> Perhaps we should have a parameter to set this (in [moby/moby#42872](https://github.com/moby/moby/pull/42872), I used a `PACKAGER_NAME` arg for this). This way (e.g.) Microsoft / Azure packaging / building the binaries would be able to set their name.

Sure I can change that.

> "company that produced the file"

Means the binary for me.

Reply:
> Perhaps we should have a parameter to set this (in [moby/moby#42872](https://github.com/moby/moby/pull/42872), I used a `PACKAGER_NAME` arg for this). This way (e.g.) Microsoft / Azure packaging / building the binaries would be able to set their name.

Done can be used:

```shell
COMPANY_NAME=Microsoft docker buildx bake --set "*.platform=windows/amd64"
# or
GOOS=windows GOARCH=amd64 COMPANY_NAME=Microsoft ./scripts/build/binary
```